File: Problem_20/solution_20.lua
-- Project Euler Problem 20 (Code Name: RIP Lua, using Python for big numbers)
-- Source: https://projecteuler.net/problem=20
function getFactorial(n)
	for i=1, n do
		n = n * i
	end
	return n
end
function getSumOfDigits(number)
	sum = 0
	for i=1, string.len(number) do
		sum = sum + string.sub(number,i,i)
	end
	return sum
end
print(getSumOfDigits(getFactorial(100)))

-- THIS IS JUST A CONCEPT
-- It would work however if the factorial of 100 wouldn't be that large so Lua saves it as x * 10^y in string format

File: Problem_20/solution_20.py
# Project Euler Problem 20 (Code Name: RIP Lua, using Python for big numbers)
# Source: https://projecteuler.net/problem=20

import time
start_time = time.time()

def getFactorial(n):
	for i in range(1, n):
		n = n * i
	return n

def getSumOfDigits(number):
	summe = 0
	number = str(number)
	for i in range(0, len(number)):
		summe = summe + int(number[i])
	return summe

print(getSumOfDigits(getFactorial(100)))
print("The script took %s seconds for execution" % (time.time() - start_time))

# Solution: 648 (correct)
# Runtime: 0.0 seconds